agent/engine/engine_sudo_linux_integ_test.go

```diff
@@ -164,6 +174,20 @@ func TestFirelensFluentbit(t *testing.T) {
 	taskEngine, done, _ := setup(cfg, nil, t)
 	defer done()
 
+	// Mock task metadata server as the firelens container needs to access it.
+	// Note that the listener has to be overwritten here, because the default one from httptest.NewServer
+	// only listens to localhost and isn't reachable from container running in bridge network mode.
+	l, err := net.Listen("tcp", ":0")
+	require.NoError(t, err)
+	server := httptest.NewUnstartedServer(mockTaskMeatadataHandler)
+	server.Listener = l
+	server.Start()
+	defer server.Close()
+
+	port := getPortFromAddr(server.URL, t)
```

Comment:
nit: let's past `t` first, as that seems to be the convention

Reply:
updated